src/components/Form/index.js

```diff
@@ -0,0 +1,184 @@
+import React, { Component } from 'react'
+import dot from 'dot-object'
+
+export default class Form extends Component {
+   static propTypes = {
+      children: React.PropTypes.node,
+      onError: React.PropTypes.func,
+      onSubmit: React.PropTypes.func
+   }
+
+   constructor(props) {
+      super(props)
+
+      this.handleSubmit = this.handleSubmit.bind(this)
+
+      this.initialValuesLoaded = false
+
+      // Setup state
+      this.state = {}
+      props.children.forEach((child) => {
+         if (this.childIsRelevant(child)) {
+            this.state[child.props.name] = {
+               value: '',
+               meta: {}
+            }
+         }
+      })
+   }
+
+   componentWillReceiveProps(props) {
+      let initialValues = props.initialValues
+
+      if (initialValues && !this.initialValuesLoaded) {
+         const newState = {}
+
+         initialValues = dot.dot(initialValues)
+
+         props.children.forEach((child) => {
+            if (this.childIsRelevant(child)) {
+               const value = initialValues
+                  ? initialValues[child.props.name] || ''
+                  : ''
+               newState[child.props.name] = {
+                  value,
+                  meta: {
+                     touched: false,
+                     error: this.checkError(child, value, props)
+                  }
+               }
+            }
+         })
+
+         this.initialValuesLoaded = true
+         this.setState(newState)
+      }
+   }
+
+   getTestFromString(str) {
+      switch (str) {
+         case 'required':
+            return value => !!value
+         case 'email':
+            return (value) => {
+               const regex = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/ // eslint-disable-line
```

Comment:
Changed to the W3C one, simply because the first one allow's single letter TLD's. 786660e